csrc/expr_evaluator.cpp

```diff
@@ -127,6 +128,44 @@ void validateValWithConcreteValue(
   }
 }
 
+std::vector<int64_t> unshardedSizes(
+    const TensorView* tv,
+    c10::IntArrayRef sizes) {
+  std::vector<int64_t> unsharded_sizes = sizes.vec();
+
+  for (IterDomain* alloc_id : tv->getMaybeAllocationDomain()) {
+    const ParallelType parallel_type = alloc_id->getParallelType();
+    if (!isParallelTypeDeviceDim(parallel_type)) {
+      continue;
+    }
+
+    const auto inputs = IterVisitor::getInputsTo(
+        {alloc_id},
+        {tv->getLogicalDomain().begin(), tv->getLogicalDomain().end()});
+    if (inputs.empty()) {
+      // FIXME: is this even possible? Logical ought to dominate allocation.
```

Comment:
It is possible with the new indexer. For example, we can have a 1d tensor, split it to a 2d tensor and call it the logical domain of a 2d tensor. It's just not common, so it's completely reasonable to not support it.